lib/class-wp-theme-json-gutenberg.php

```diff
@@ -4085,4 +4085,35 @@ protected static function get_valid_block_style_variations() {
 
 		return $valid_variations;
 	}
+
+	/**
+	 * Resolves relative paths in theme.json styles to theme absolute paths.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return WP_Theme_JSON_Gutenberg The current instance of $this.
+	 */
+	public function resolve_theme_file_uris() {
```

Comment:
I've tentatively added this method to `WP_Theme_JSON_Gutenberg` for now as it does pertain directly to theme.json values and also themes themselves. 

Having said that, my first instinct was to house it as a static method in `WP_Theme_JSON_Resolver_Gutenberg`, because we're resolving values. 😄 

I tested something similar and it requires passing a theme json object, getting the data from that object, updating the style values on the data, then returning a new `WP_Theme_JSON_Resolver_Gutenberg`.

This would however mean that, wherever stylesheets are being generated for the editor e.g., editor settings or the class-wp-rest-global-styles-controller.php, Gutenberg/WordPress would have insert the extra step of `WP_Theme_JSON_Resolver_Gutenberg::resolve_theme_file_uris()`.

Still, I can imagine that theme devs may just want to pass a theme.json config to `WP_Theme_JSON_Gutenberg` and have the option to resolve URLs directly, e.g., `$theme_json->resolve_theme_file_uris()->get_stylesheet()` - the other option is that we do it all under the hood, so no extra step is required, but that might be limiting in case relative paths have other uses. It would be retain the imperative nature as well.

cc @oandregal or @youknowriad for thoughts

Reply:
I'd rather do this elsewhere — for example, the resolver. A good guidance is: if a change couples the `WP_Theme_JSON` class with any WordPress utility (theme supports, get_theme_file_uri, etc.) shouldn't be here.

For this specific case, one way to do this could be to create the `resolve_theme_file_uris` private method in the resolver and use it like this:

```php
$result = $this->resolve_theme_file_uris( $result );
```

Essentially, in the same places that it's been called now, but implemented in the resolver instead of the theme.json class.

Would that work for you?

Reply:
Thanks for the feedback @oandregal! 🙇🏻 

Yes, that would work in this case I believe - as mentioned above, it was the first thing I tried and it worked. I guess I got lost in semantics again 🙃 

CheersQ